tests/end_to_end/end_to_end.cpp

```diff
@@ -522,10 +533,9 @@ divide_by_zero_test_um(ebpf_execution_type_t execution_type)
 
     REQUIRE(hook_result == 0);
 
-    hook.detach_link(link);
-    hook.close_link(link);
+    hook.detach_and_close_link(&link);
 
-    bpf_object__close(object);
+    bpf_object__close(unique_object.release());
```

Comment:
Genuine question: why couldn't we use a custom deleter for this unique ptr to do the close for us at its destruction?
(The current implementation is fine. Even if a custom deleter _is_ possible, we can look at it in the next implementation update)

Reply:
We are using custom deleter for this unique_ptr (see line 78 in this file). In the case the test bails due to a failure, custom deleter will clean up the `bpf_object`.

I agree we can completely remove line 538 and it will still work, but I wanted to keep the flow similar to what it was earlier.